Lib/_pyio.py

```diff
@@ -978,6 +978,13 @@ def tell(self):
             raise ValueError("tell on closed file")
         return self._pos
 
+    def peek(self, size=1):
+        if self.closed:
+            raise ValueError("peek on closed file")
+        if size < 1:
+            size = len(self._buffer) - self._pos
```

Comment:
The C implementation is differently implemented, which surprised me at first.

Reply:
Do you mean that you would have expected something like `if size < 1 or size > len(self._buffer)`? The Python version relies on how slicing semantics work to deal with this. Shall I change it to be more explicit?

Reply:
I don't think that is needed. Perhaps a comment?

Reply:
I added a comment. Let me know if it’s too verbose now.